sdk/core/System.ClientModel/src/Convenience/KeyCredential.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+
+namespace System.ClientModel;
+
+public class KeyCredential
+{
+    private string _key;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KeyCredential"/> class.
+    /// </summary>
+    /// <param name="key">Key to use to authenticate with the Azure service.</param>
+    /// <exception cref="System.ArgumentNullException">
+    /// Thrown when the <paramref name="key"/> is null.
+    /// </exception>
+    /// <exception cref="System.ArgumentException">
+    /// Thrown when the <paramref name="key"/> is empty.
+    /// </exception>
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+    public KeyCredential(string key) => Update(key);
+#pragma warning restore CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+
+    // Note: this is GetValue instead of GetKey to allow consistent naming
+    // across credential types.  For example, for NamedKeyCredential, we would have
+    // GetValues with two out params to enable atomicity in reading credential values.
+    public string GetValue() => Volatile.Read(ref _key);
```

Comment:
In the Azure.Core types I think we purposely didn't expose methods/properties for the key values due to security concerns. Is this not a concern?

Reply:
Discussed offline - we will move to [Deconstruct APIs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/AzureNamedKeyCredential.cs#L73) here, as it will give us consistency across credential types as new types like NamedKeyCredential come online, pending a strong concern from the BCL team (Anne to dig up context on the prior concern).